paddle/fluid/operators/elementwise/elementwise_pow_op.h

```diff
@@ -12,17 +12,59 @@ limitations under the License. */
 #pragma once
 
 #include <cmath>
+#include <type_traits>
 #include "paddle/fluid/operators/elementwise/elementwise_op.h"
 #include "paddle/fluid/operators/elementwise/elementwise_op_function.h"
 
 namespace paddle {
 namespace operators {
 
-template <typename T>
-struct PowFunctor {
+namespace details {
+
+// Only valid if T is an integral type.
+template <class T,
+          class = typename std::enable_if<std::is_integral<T>::value>::type>
+static inline HOSTDEVICE T ipow(T base, T exp) {
```

Comment:
`static_assert(std::is_integral<T>::type, "T must be integral");`, and remove `std::enable_if`?

Reply:
Thanks, implementation changed.